libs/widget-lib/docs/README.md

```diff
@@ -75,99 +74,57 @@ An example of connecting a widget to Metamask:
+| `height`               | `number`                | The height of the widget in pixels. Default: 600px                                                                                                           |
+| `metaData`             | `CowSwapWidgetMetaData` | Information about the application in which the widget is embedded. This information will help identify the source of orders and requests from users.         |
+| `provider`             | `EthereumProvider`      | The Ethereum provider to be used for interacting with a wallet.                                                                                              |
+| `chainId`              | `number`                | The blockchain ID on which the trade will take place.                                                                                                        |
+| `tradeType`            | `string`                | The type of trade. Can be `swap` or `limit-orders`.                                                                                                          |
+| `env`                  | `CowSwapWidgetEnv`      | The environment of the widget (`'local'` or `'prod'`).                                                                                                       |
+| `tradeAssets`          | `TradeAssets`           | An object containing information about the selling and buying assets.                                                                             |
+| `theme`                | `CowSwapTheme`          | The theme of the widget (`'dark'` for dark theme or `'light'` for light theme).                                                                   |
+| `logoUrl`              | `boolean`               | The width of the widget in pixels.                                                                                                                |
+| `hideLogo`             | `boolean`               | The height of the widget in pixels.                                                                                                               |
+| `hideNetworkSelector`  | `boolean`               | Disables an opportunity to change the network from the widget UI.                                                                                 |
+| `dynamicHeightEnabled` | `boolean`               | Dynamically changes the height of the iframe depending on the content.                                                                            |
+| `enabledTradeTypes`    | `Array<TradeType>`      | CowSwap provides three trading widgets: swap, limit and twap orders. Using this option you can narrow down the list of available trading widgets. |
+| `palette`              | `CowSwapWidgetPalette`  | Using the palette you can customize the appearance of the widget. For example, you can change the main color of the background and text.          |
```

Comment:
I'm afraid we need both.
`them` sets a general subset of colors.
`palette` only overrides some of them

Reply:
yep, but i could see that in the future `palette` has more and more options. 

BEcause both are part of `theming`, as a dev i find it easier to find if they show close to each other so their names are similar.


So i was suggesting leaving both: `theme` & `themePallete` 

However, there's another alternative, since apps will likely setup this once. 

What about defining it like this?

```ts

type ThemeBase = 'light' | 'dark' | 'auto'

export interface CowSwapTheme {
  baseTheme: ThemeBase
  primaryColor: string
  screenBackground: string
  widgetBackground: string
  textColor: string
}
```

Then, the theme type would be
```ts
{
    theme: ThemeBase | CowSwapTheme
    ...
}

// so users can do:
theme = 'light'
theme = 'dark'
theme = {
   baseTheme: 'light,
   primaryColor: '#fff'
   ...
}
```

Alternatively:
```ts
type ThemeBase = 'light' | 'dark' | 'auto'

export interface CowSwapTheme {
  baseTheme?: ThemeBase
  primaryColor?: string
  screenBackground?: string
  widgetBackground?: string
  textColor?: string
}

{
    theme?: CowSwapTheme // defaults to "{ baseTheme: 'auto' }" which uses the OS
    ...
}
```

Reply:
Got your point.
Don't you mind if I will implement this in the next PR?
Anyway, currently the palette just randomly bound to colors. It needs and input from Michel